modules/datadict/jsx/dataDictIndex.js

```diff
@@ -14,71 +16,38 @@ import Loader from 'Loader';
  * @version 1.0.0
  *
  * */
-class DataDictIndex extends React.Component {
+class DataDictIndex extends Component {
   constructor(props) {
     super(props);
 
     this.state = {
+      data: {},
+      error: false,
       isLoaded: false,
-      filter: {},
-      hiddenHeaders: [],
     };
 
-    /**
-     * Set filter to the element's ref for filtering
-     */
-    this.filter = null;
-    this.setFilterRef = (element) => {
-      this.filter = element;
-    };
-
-    /**
-     * Bind component instance to custom methods
-     */
     this.fetchData = this.fetchData.bind(this);
-    this.updateFilter = this.updateFilter.bind(this);
-    this.resetFilters = this.resetFilters.bind(this);
     this.formatColumn = this.formatColumn.bind(this);
   }
 
   componentDidMount() {
-    this.fetchData();
+    this.fetchData()
+      .then( () => this.setState({isLoaded: true}));
   }
 
   /**
    * Retrive data from the provided URL and save it in state
-   * Additionaly add hiddenHeaders to global loris variable
-   * for easy access by columnFormatter.
+   *
+   * @return {object}
    */
   fetchData() {
-    $.ajax(this.props.DataURL, {
-      method: 'GET',
-      dataType: 'json',
-      success: (data) => {
-        this.setState({
-          data: data,
-          isLoaded: true,
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+        .then((resp) => resp.json())
+        .then((data) => this.setState({data}))
+        .catch((error) => {
```

Comment:
On a side note: This will catch error thrown by fetch, not HTTP 4xx/5xx. But conveniently, since `DataFrameworkMenu` extends `NDB_Page`, when an error happen (4xx/5xx) an html page is return and the `resp.json()` throws a TypeError because the html can't be parsed.

Reply:
That doesn't sound like a reliable side-effect to depend on.